opensquirrel/passes/router/shortest_path_router.py

```diff
@@ -0,0 +1,34 @@
+import networkx as nx
+
+from opensquirrel.ir import IR, SWAP, Gate
+from opensquirrel.passes.router import Router
+
+
+class ShortestPathRouter(Router):
+    def __init__(self, connectivity: dict[str, list[int]]) -> None:
+        self.connectivity = connectivity
+
+    def route(self, ir: IR) -> IR:
+        """
+        Routes the circuit by inserting SWAP gates to make it executable given the hardware connectivity.
+        Args:
+            ir: The intermediate representation of the circuit.
+        Returns:
+            The intermediate representation of the routed circuit (including the additional SWAP gates).
+        """
+        graph = nx.Graph({int(k): [int(v) for v in values] for k, values in self.connectivity.items()})
+        instruction_counter = 0
```

Comment:
Instruction is only a specific kind of statement and you are going through a list of statements. I would use a 'counter' if I would actually what to know the sum total of something (at some point), but here it is used as an index.

```suggestion
        statement_index = 0
```